### PR TITLE
DRILL-5051: Fix incorrect result returned in nest query with offset specified

### PR DESCRIPTION
**Pull request by zbdzzg**
1. Fix this problem.
2. Merge two limit methods, most of code is the same.

### CODE REVIEW

**Comment by zbdzzg** [comment]
@sudheeshkatkam Could you please review this? Thanks.

**Comment by sudheeshkatkam** [comment]
+1

Can you please change the commit message to "DRILL-5051: Fix incorrect computation of 'fetch' in LimitRecordBatch when 'offset' is specified"? The JIRA/PR title is fine as is.

**Comment by zbdzzg** [comment]
@sudheeshkatkam Thanks for your review, commit message has been changed.

**Comment by sudheeshkatkam** [comment]
@zbdzzg Thank you for the patch!

File: exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java [hunk 1]
```diff
@@ -221,4 +221,14 @@ public void testDRILL4884() throws Exception {
             .baselineRecords(baseline)
             .go();
   }
+
+  @Test
+  public void testDRILL5051() throws Exception {
+    testBuilder()
+        .sqlQuery("select count(1) as cnt from (select l_orderkey from (select id from cp.`tpch/lineitem.parquet` limit 2) limit 1 offset 1)")
```
Comment (jinfengni) [inline exec/java-exec/src/test/java/org/apache/drill/TestBugFixes.java]: This query seems to hit SQL validation error " Column 'l_orderkey' not found in any table". 

I'm going to change column "id" to "l_orderkey" in the inner subquery, in order to skip this validation error.

**Pull request closed without merging**